### PR TITLE
Don't allow Reset current property pad in case of item focused changed

### PR DESCRIPTION
**Pull request by netonjm**
Fixes #999288 - Native Property Pad switches context when it loses focus

### CODE REVIEW

File: main/src/addins/MonoDevelop.DesignerSupport/MonoDevelop.DesignerSupport/PropertyPadVisitor.cs [hunk 1]
```diff
@@ -73,7 +73,6 @@ public void End ()
 						return;
 					}
 				}
-				DesignerSupport.Service.ReSetPad ();
```
Comment (sevoku) [inline main/src/addins/MonoDevelop.DesignerSupport/MonoDevelop.DesignerSupport/PropertyPadVisitor.cs]: I'm not sure why it was required here, it doesn't make much sense to me. @mkrueger or @slluis do you remember (this is more than 7 years old code)? Is it safe do remove this call here, or are we missing something?

Comment (Therzok) [reply main/src/addins/MonoDevelop.DesignerSupport/MonoDevelop.DesignerSupport/PropertyPadVisitor.cs]: Seems to be done every time before a property pad needs to be refreshed:
http://source.monodevelop.com/#MonoDevelop.DesignerSupport/MonoDevelop.DesignerSupport/DesignerSupportService.cs,2e5f832abcff8e71,references

Comment (netonjm) [reply main/src/addins/MonoDevelop.DesignerSupport/MonoDevelop.DesignerSupport/PropertyPadVisitor.cs]: Yeah, I'm not sure if it was to cover some specific case (which I did not find) when the property panel needs to be reseted, or maybe it's an old code.

**Comment by sevoku** [comment]
@monojenkins backport release-8.4

File: main/src/addins/MonoDevelop.DesignerSupport/MonoDevelop.DesignerSupport/PropertyPadVisitor.cs (hunk 1 above)
Comment (sevoku) [reply main/src/addins/MonoDevelop.DesignerSupport/MonoDevelop.DesignerSupport/PropertyPadVisitor.cs]: yeah, that's clear, makes sense, but not really after walking the chain and not finding anything. This behaviour can be also observed with the Gtk pad (especially if it's autohiding), so it's not a regression with proppy.

**Comment by Therzok** [comment]
@sevoku @slluis this PR introduced leaks in the IDE

https://dev.azure.com/DevDiv/DevDiv/_workitems/edit/1000899

Please prioritize, as it breaks CI

**Pull request merged**